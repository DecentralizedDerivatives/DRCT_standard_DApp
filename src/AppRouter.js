import React, { Component } from 'react';
import { Switch, Route, BrowserRouter as Router } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import 'react-dates/initialize';

import ConnectionModal from './components/ConnectionModal';
import BlockProgress from './components/BlockProgress';
import Header from './components/Header';
import Landing from './components/Landing';
import MyPortfolio from './components/MyPortfolio';
import Bulletin from './components/Bulletin';
import HowTo from './components/HowTo';
import Terms from './components/Terms';
import { checkUserConnection } from './actions/statusActions';
import './style.css';

class AppRouter extends Component {
  constructor() {
    super();
    this.state = { showTerms: false };
  }
  componentDidMount() {
    this.props.checkUserConnection();
  }

  componentWillUpdate() {
    this.props.checkUserConnection();
  }

  render() {
    return (
      <Router>
        <div className="app">
          <Header
            showTerms={() => { this.setState({showTerms: Boolean(true)}); }}
            isConnected={this.props.metamask && this.props.network === require('./config/keys').network_id}
            whiteListed={this.props.whiteListed}
          />
          <Route exact path="/" component={Landing} />
          <Switch>
            <Route exact path="/portfolio" component={MyPortfolio} />
            <Route path="/bulletin" component={Bulletin} />
            <Route path="/how-to" component={HowTo} />
          </Switch>

          <BlockProgress />
          <ConnectionModal />
          { this.state.showTerms ?
            (
              <Terms close={() => { this.setState({showTerms: Boolean(false)}); }} />
            ) : ''}
        </div>
      </Router>
    );
  }
}

AppRouter.propTypes = {
  checkUserConnection: PropTypes.func.isRequired,
  metamask: PropTypes.bool.isRequired,
  whiteListed: PropTypes.bool.isRequired,
  network: PropTypes.number.isRequired
};

const mapStateToProps = state => ({
  metamask: state.status.connectStatus.metamask,
  network: state.status.connectStatus.network,
  whiteListed: state.status.connectStatus.whiteListed
});

export default connect(
  mapStateToProps,
  { checkUserConnection }
)(AppRouter);
