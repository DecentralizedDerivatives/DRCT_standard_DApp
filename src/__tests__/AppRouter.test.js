import AppRouter from '../AppRouter';

describe('<AppRouter />', () => {
  it('renders the component', () => {
    const wrapper = shallow(<AppRouter store={initStore()} />).dive();

    expect(wrapper).toMatchSnapshot();
  });

  it('renders the component with data', () => {
    const wrapper = shallow(<AppRouter store={initStore(FIXTURE)} />).dive();

    wrapper.setState({});

    wrapper.find('Header').prop('showTerms')();
    wrapper.find('Terms').prop('close')();

    expect(wrapper).toMatchSnapshot();
  });
});
