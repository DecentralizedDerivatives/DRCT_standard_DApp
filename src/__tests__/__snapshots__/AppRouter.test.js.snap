// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppRouter /> renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppRouter
    checkUserConnection={[Function]}
    metamask={false}
    network={0}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    whiteListed={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="app"
      >
        <Header
          isConnected={false}
          showTerms={[Function]}
          whiteListed={false}
        />
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/portfolio"
          />
          <Route
            component={[Function]}
            path="/bulletin"
          />
          <Route
            component={[Function]}
            path="/how-to"
          />
        </Switch>
        <Connect(BlockProgress) />
        <Connect(ConnectionModal) />
        
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            isConnected={false}
            showTerms={[Function]}
            whiteListed={false}
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/portfolio"
            />
            <Route
              component={[Function]}
              path="/bulletin"
            />
            <Route
              component={[Function]}
              path="/how-to"
            />
          </Switch>,
          <Connect(BlockProgress) />,
          <Connect(ConnectionModal) />,
          "",
          undefined,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isConnected": false,
            "showTerms": [Function],
            "whiteListed": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/portfolio"
              />,
              <Route
                component={[Function]}
                path="/bulletin"
              />,
              <Route
                component={[Function]}
                path="/how-to"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/portfolio",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/bulletin",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/how-to",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "",
        undefined,
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="app"
        >
          <Header
            isConnected={false}
            showTerms={[Function]}
            whiteListed={false}
          />
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/portfolio"
            />
            <Route
              component={[Function]}
              path="/bulletin"
            />
            <Route
              component={[Function]}
              path="/how-to"
            />
          </Switch>
          <Connect(BlockProgress) />
          <Connect(ConnectionModal) />
          
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Header
              isConnected={false}
              showTerms={[Function]}
              whiteListed={false}
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/portfolio"
              />
              <Route
                component={[Function]}
                path="/bulletin"
              />
              <Route
                component={[Function]}
                path="/how-to"
              />
            </Switch>,
            <Connect(BlockProgress) />,
            <Connect(ConnectionModal) />,
            "",
            undefined,
          ],
          "className": "app",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isConnected": false,
              "showTerms": [Function],
              "whiteListed": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/portfolio"
                />,
                <Route
                  component={[Function]}
                  path="/bulletin"
                />,
                <Route
                  component={[Function]}
                  path="/how-to"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/portfolio",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/bulletin",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/how-to",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "",
          undefined,
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<AppRouter /> renders the component with data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AppRouter
    checkUserConnection={[Function]}
    metamask={true}
    network={4}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    whiteListed={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div
        className="app"
      >
        <Header
          isConnected={true}
          showTerms={[Function]}
          whiteListed={true}
        />
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/portfolio"
          />
          <Route
            component={[Function]}
            path="/bulletin"
          />
          <Route
            component={[Function]}
            path="/how-to"
          />
        </Switch>
        <Connect(BlockProgress) />
        <Connect(ConnectionModal) />
        
        <Redirect
          push={false}
          to="/bulletin"
        />
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            isConnected={true}
            showTerms={[Function]}
            whiteListed={true}
          />,
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/portfolio"
            />
            <Route
              component={[Function]}
              path="/bulletin"
            />
            <Route
              component={[Function]}
              path="/how-to"
            />
          </Switch>,
          <Connect(BlockProgress) />,
          <Connect(ConnectionModal) />,
          "",
          <Redirect
            push={false}
            to="/bulletin"
          />,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "isConnected": true,
            "showTerms": [Function],
            "whiteListed": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "component": [Function],
            "exact": true,
            "path": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/portfolio"
              />,
              <Route
                component={[Function]}
                path="/bulletin"
              />,
              <Route
                component={[Function]}
                path="/how-to"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/portfolio",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/bulletin",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "path": "/how-to",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "push": false,
            "to": "/bulletin",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div
          className="app"
        >
          <Header
            isConnected={true}
            showTerms={[Function]}
            whiteListed={true}
          />
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/portfolio"
            />
            <Route
              component={[Function]}
              path="/bulletin"
            />
            <Route
              component={[Function]}
              path="/how-to"
            />
          </Switch>
          <Connect(BlockProgress) />
          <Connect(ConnectionModal) />
          
          <Redirect
            push={false}
            to="/bulletin"
          />
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Header
              isConnected={true}
              showTerms={[Function]}
              whiteListed={true}
            />,
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />,
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/portfolio"
              />
              <Route
                component={[Function]}
                path="/bulletin"
              />
              <Route
                component={[Function]}
                path="/how-to"
              />
            </Switch>,
            <Connect(BlockProgress) />,
            <Connect(ConnectionModal) />,
            "",
            <Redirect
              push={false}
              to="/bulletin"
            />,
          ],
          "className": "app",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "isConnected": true,
              "showTerms": [Function],
              "whiteListed": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "component": [Function],
              "exact": true,
              "path": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/portfolio"
                />,
                <Route
                  component={[Function]}
                  path="/bulletin"
                />,
                <Route
                  component={[Function]}
                  path="/how-to"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/portfolio",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/bulletin",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "path": "/how-to",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "push": false,
              "to": "/bulletin",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
